phoenix-core/src/main/java/org/apache/phoenix/query/QueryServicesOptions.java

```diff
@@ -355,7 +355,7 @@
     public static final long DEFAULT_TASK_HANDLING_MAX_INTERVAL_MS = 30*60*1000; // 30 min
     public static final long DEFAULT_TASK_HANDLING_INITIAL_DELAY_MS = 10*1000; // 10 sec
 
-    public static final long DEFAULT_GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS = 10*60*1000; /* 10 minutes */
+    public static final long DEFAULT_GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS = 7*24*60*60*1000; /* 7 days */
```

Comment:
How did we arrive on 7 days as the stopgap before deleting unverified rows? Is it related to the frequency of major compaction or just a general "enough time" to ensure replication lag is covered?

If the former, I would suggest to set this to the value set for the config HConstants.MAJOR_COMPACTION_PERIOD` or default to `HConstants.DEFAULT_MAJOR_COMPACTION_PERIOD`.

Reply:
The team including Geoffrey and Vincent decided that 7 day delete time is long enough in practice to make sure a row will be replicated eventually.